### PR TITLE
Update course completion metrics API

### PR DESCRIPTION
**Pull request by msaqib52**
This PR updates course completion metrics API to allow filtering of users by groups and disables default limit of three on leaders, also did some refactoring in this PR.

@ziafazal 👓

### CODE REVIEW

File: lms/djangoapps/progress/models.py [hunk 1]
```diff
@@ -41,8 +41,9 @@ def get_total_completions(cls, course_key, exclude_users=None, org_ids=None):
             .exclude(user__id__in=exclude_users)
         if org_ids:
             queryset = queryset.filter(user__organizations__in=org_ids)
-        completions = queryset.aggregate(total=Sum('completions'))
-        completions = completions['total'] or 0
+        if group_ids:
+            queryset = queryset.filter(user__groups__in=group_ids).distinct()
+        completions = sum([student_progress.completions for student_progress in queryset])
```
Comment (ziafazal) [inline lms/djangoapps/progress/models.py]: What is the reason for using list comprehension here instead of `queryset.aggregate`?

**Comment by ziafazal** [comment]
@msaqib52 Please add unit tests also to support your changes. Once you have written tests you can view coverage by running `paver coverage` to make sure you have >90% test coverage of you changes.

**Comment by ziafazal** [comment]
LGTM 👍

File: lms/djangoapps/progress/models.py (hunk 1 above)
Comment (msaqib52) [inline lms/djangoapps/progress/models.py]: There is a bug in older versions of django(< 1.8), Aggregate Sum doesn't respect distinct() in the queryset filter. https://code.djangoproject.com/ticket/24385

**Pull request merged**